### ISSUE TITLE
No mobile points

### ISSUE THREAD

**Issue by krizzo** [justinxxvii/python-bing-bot#11]
I'm getting 14/15 desktop and 0/10 mobile points.

**Comment by rockylikespocky** [comment]
Have you tried running it again? Sometimes my net gets wonky and the bot times out. Everything seems to be working for me except for the point recording, that is a hit or miss.

**Comment by krizzo** [comment]
It's not the timeout issue it successfully exits after attempting the 30 searches. but it reports, and I confirmed, desktop 14/15 and mobile 0/10. I'll dig into the script tonight when I get home see what's happening.

**Comment by justinxxvii** [comment]
The bot bases its searches on detecting the "Earn X credits per X Bing
searches up to X credits a day". So it is correctly identifying the
multiplier and the maximum. At the end of the run it opens the rewards pane
and scans again, so not sure why it would be detecting 14 and not 15. Are
you saying that it only earns you 14 credits, or does it earn all 15 and
just say 14?

**Comment by krizzo** [comment]
It's correctly detecting the rewards at the end. It's only getting 14/15 and none of the mobile ones. I've tried with multiple accounts, single accounts different IPs (locations). I'll test it again maybe it was a fluke with my network connecting to MS.

**Comment by justinxxvii** [comment]
There's no way to pause, sorry for that. When I troubleshoot I start adding
print lines at pretty much every step of the code. Do you know any python,
you could try adding some print statements to see what variables are being
changed and updated at each step in the PBBDetector files. I just looked at
my console from yesterday and I had 15/15 and 10/10

**Comment by justinxxvii** [comment]
Only other thing I can think of is if you want to take a screenshot of your
rewards pane, redact all necessary info and just paste a link to a dropbox
file or something so we can compare. Every once in awhile it seems like
different logout mechanisms start rolling out. They recently changed the
freebie credit which broke that process, (haven't pushed the fill fix yet)

**Comment by krizzo** [comment]
Yeah I'll add some logger debug and info lines to help. If I still get get things working I'll do that doc showing the results. Thanks, glad to know yous worked so it's most likely something on my end.

**Comment by Gishi1** [comment]
iv been having a problem with the browser some times loads for ever and it ends up erroring.  but if i run the scrip as admin it runs fine... i just dont get it.

**Comment by justinxxvii** [comment]
I usually force load chrome and then quit before running the script. Most of the launch crashes are because the script is timing out trying to connect to the selenium process 


    _____________________________

**Comment by Gishi1** [comment]
well i have phantomjs having the same problem so i don't think its the drivers i think its bing not seeing the bot as being logged in. because if i set a delay for 15 seconds before the bot grabs the point info from the pane it seems to load the data. i think the only way to fix this is to make the bot check bing for a log in status.

**Comment by Gishi1** [comment]
from bad to worse now its not seeing me as logged in no matter how long i have it sleep.

**Comment by justinxxvii** [comment]
The points detection is only bound to login status so far that your points
totals are only displayed if you are logged in. Do this: set the sleep
delay to a very high number, and navigate to the rewards page yourself.
Screenshot the list of daily rewards, redact any identifying information,
and share it.

If it's not detecting points it's because the functions in the script
aren't following the correct rabbit hole. It works by finding a specific
string of text to identify the "mobile section" (Find "Mobile search - ")
and then doing some math to determine how many total characters are
present, traversing over a certain distance to be able to get the "X of X
credits" string and identify the min and max numbers. If any of this fails
or if strings are encountered that are not expected the whole thing will
fail.

I'm wondering if there is a rolling change or experimental update that some
bing members are beta testing, that has slightly altered the wording. If
you sit and watch the script run, and it goes past the Outlook screen to
the bing homepage, and you can click on the rewards list and the
login/logout link and it shows that you're logged in, then it's not the
login or sleep process.

**Comment by Gishi1** [comment]
well i used this and it works flawlessly for mobile:

    def check_mobile_login(self):
        print "Checking mobile login"
        self.browser.get('http://www.bing.com/rewards/dashboard')
        self.browser.find_element_by_xpath('//*[@id="mbHeader"]/a[2]/img').click()
        self.browser.find_element_by_xpath('//*[@id="Account"]/img').click()
        if self.browser.find_element_by_partial_link_text('Sign in') > -1:
           print "Forcing a login to mobile"
           self.browser.find_element_by_partial_link_text('Sign in').click()
